docs/designs/sqlflow_integration_design.md

```diff
@@ -0,0 +1,237 @@
+# Design Overview for ElasticDL on SQLFlow
+
+## Overview
+
+This is a design doc on integration with [SQLFlow](https://github.com/sql-machine-learning/sqlflow).
+
+### User Interface
+
+#### Training Job Submission
+
+```sql
+SELECT
+    c1, c2, c3, c4, c5 as class
+FROM training_data
+TRAIN ElasticDLKerasClassifier
+WITH
+    model.optimizer = "optimizer",
+    model.loss = "loss",
+    model.eval_metrics_fn = "eval_metrics_fn",
+    model.num_classes = 3,
+    model.dataset_fn = "dataset_fn",
+    train.shuffle = 120,
+    train.epoch = 2,
+    train.grads_to_wait = 2,
+    train.tensorboard_log_dir = "",
+    train.checkpoint_steps = 0,
+    train.checkpoint_dir = "",
+    train.keep_checkpoint_max = 0,
+    eval.steps = 0,
+    eval.start_delay_secs = 100,
+    eval.throttle_secs = 0,
+    eval.checkpoint_filename_for_init = "",
+    engine.master_resource_request = "cpu=400m,memory=1024Mi",
+    engine.master_resource_limit = "cpu=1,memory=2048Mi",
+    engine.worker_resource_request = "cpu=400m,memory=2048Mi",
+    engine.worker_resource_limit = "cpu=1,memory=3072Mi",
+    engine.num_workers = 2,
+    engine.volume = "",
+    engine.image_pull_policy = "Never",
+    engine.restart_policy = "Never",
+    engine.extra_pypi_index = "",
+    engine.namespace = "default",
+    engine.minibatch_size = 64,
+    engine.master_pod_priority = "",
+    engine.cluster_spec = "",
+    engine.num_minibatches_per_task = 2,
+    engine.docker_image_repository = "",
+    engine.envs = ""
+COLUMN
+  c1,
+  c2,
+  c3,
+  c4
+LABEL class
+INTO trained_elasticdl_keras_classifier;
+```
+
+#### Prediction Job Submission
+
+```sql
+SELECT
+    c1, c2, c3, c4
+FROM prediction_data
```

Comment:
How to make sure the prediction has the same input name as the training?

Reply:
> How to make sure the prediction has the same input name as the training?

Good question. Seems that SQLFlow needs to know the model input names so that it can generate `dataset_fn` with correct input names.

Reply:
@tonyyang-svail How does SQLFlow currently handle this? Are feature names saved somewhere as part of training?

Reply:
At the end of the training, SQLFlow saves the train SQL along with the model files. In a prediction job, the train SQL will be used to determine the input schema.